### DISCUSSION TITLE
ESP32 dev kit - how to reset wifi settings?

### DISCUSSION THREAD

**Discussion by Frank-Bemelman** [arendst/Tasmota#12504] [Q&A] [answered]
Flashed an ESP32 dev kit with Tasmota 9.5.0 and it runs and behaves as expected. I was surprised I managed that.

I noticed that resetting the wifi settings of the module does not seem to work. The usual method is to press a button for 40 seconds, or to cycle power for 6 (7) times. I don't know if this also applies to the buttons on the ESP32 dev kit board, none of the two seem to do any reset of the wifi settings, Cycling power also did not have any effect.

Is there another method, other than reflashing the entire ESP32, or is this just the way it is, at the moment.

Thank you for any pointers.

Best regards,
Frank

**Comment by barbudor** [comment, marked as answer]
Hi Frank
The 40 seconds works only on gpio defined as 'Button 1' (or _i _ni variants) not any other Buttons or Switch.

The power-cycle method must be a full power cycle. If your dev board is powered by USB, you need to unplug, wait, replug, wait... This one could be disabled by 'setoption65'. 

I will try later

**Reply by barbudor** [reply]
Just tested that both options are working as expected.
However I had to power-cycle 10 times to be sure the 2nd option worked.

**Reply by barbudor** [reply]
Doing the power cycle more carefully while monitoring serial logs, I can confirm that the configuration is reset on the 7th power-up

**Reply by Frank-Bemelman** [reply]
Hi Barbudor,

Wow, that was a quick reply ;-) The 40 seconds method now works. In the Tasmota web interface, configure, configure module, set GPIO0 as button 1, and voila! The BOOT button on the ESP32 devkit module is apparently wired to GPIO0 and it now worked to reset the module. 

I did try the power cycle by unpugging the USB, with 5 seconds interval off/on  - but for some unknown reason I had no succes with that. 

Thanks a lot! I am not very experienced with Tasmota, and not sure if there is a method to reset only the wifi setup without restoring all other settings such as the GPIO pins, MQTT and what have you. But this is good enough for now. 

Thanks again,
Cheers, Frank

**Reply by Jason2866** [reply]
Take a look in the documentation section commands.
The commands to search for are `Password` and `Ssid`

**Reply by barbudor** [reply]
@Frank-Bemelman While there is no build-in recovery mechanism to erase the Wifi settings, you can combine the commands listed by Jason above within a Rule triggered by (for example) Button_2 hold for 40 seconds to execute those commands and restart.
Check the section on Button-and-Switches and Rules to understand how to use Rules to capture such button events

**Reply by sfromis** [reply]
In addition to the methods with button press or power cycling, you can also use a serial terminal program via USB for the serial console, allowing you to see what happens, and also issue commands to reset, set ssid/password, or whatever. Obviously this will not work if you config the serial pins for other purposes, but does of course not depend on wifi.

**Reply by garudaonekh** [reply]
My 6 restart cycle works But the 40seconds Button1 doesn't work for me, which is what I need. I will disable 6 restart cycle to prevent power issues cause reset by accident.
Here's what I try:
1. I set GPIO0 to Button1. Not working
2. I use 4.7K resistor to GPIO2 and set GPIO2 as Button1 also not working for me.

I am using ESP32-WROOM 32U and tasmot 9.4.

**Reply by sfromis** [reply]
How did you configure? Some SetOption settings will affect whether the 40-second reset can work or not.

**Reply by garudaonekh** [reply]
`setoption
11:30:10.380 SRC: WebConsole from 192.168.8.100
11:30:10.382 CMD: Group 0, Index 1, Command "SETOPTION", Data ""
11:30:10.386 RSL: RESULT = {"SetOption1":"OFF"}`

Configure Module->GPIO2->Button [1]

**Reply by sfromis** [reply]
I did not ask for `SetOption1` but all your config, there are *many* different SetOption values

**Reply by garudaonekh** [reply]
[config.txt](https://github.com/arendst/Tasmota/files/6745172/config.txt)
Hi, here's the config file dumped from tasmota-decode

**Reply by garudaonekh** [reply]
<img width="541" alt="Screen Shot 2021-07-01 at 10 49 48 AM" src="https://user-images.githubusercontent.com/2074857/124061979-16d67f80-da5a-11eb-888b-305f6c3bfbe7.png">
and here's my configuration in WebUI

**Reply by sfromis** [reply]
I merely meant what config *you did*, instead of just dumping it all. Beware that your config.txt file include credentials.

**Reply by garudaonekh** [reply]
As shown on my screen image, I don't set anything except this GPIO2->Button[1].

**Reply by sfromis** [reply]
That partial screen dump tells nothing about what other config you may have done, including using various commands.
